### PR TITLE
fix with suite_socialbot and support parallel load

### PR DESCRIPTION
**Pull request by witwolf**
#83

### CODE REVIEW

**Comment by witwolf** [comment]
lock handle is  incorrect previous, and now fixed

File: alf/environments/suite_socialbot.py [hunk 1]
```diff
@@ -129,19 +129,22 @@ def _get_unused_port(start, end=65536):
...
     """
     process_lock = None
+    unused_port = None
     try:
-        unused_port = None
         for port in range(start, end):
+            process_lock = InterProcessLock(
+                path='/tmp/socialbot/{}.lock'.format(port))
+            if not process_lock.acquire(blocking=False):
+                process_lock = None
```
Comment (hnyu) [inline alf/environments/suite_socialbot.py]: Line 138 & 147 are probably redundant.

**Comment by witwolf** [comment]
parallel load is not available  with SocialBot due to some unknown problems with gazebo load world file

File: alf/environments/suite_socialbot.py (hunk 1 above)
Comment (witwolf) [reply alf/environments/suite_socialbot.py]: No redundancy here  .

**Comment by hnyu** [comment]
> parallel load is not available with SocialBot due to some unknown problems with gazebo load world file

If this is the case, is it still necessary for us to change from `BatchedPyEnvironment` to `ParallelPyEnvironment`?

**Comment by hnyu** [comment]
> parallel load is not available with SocialBot due to some unknown problems with gazebo load world file

So the changes in suite_socialbot.py are still necessary if parallel loading is not available? What will be needed to fix the gazebo world file loading issue?

**Comment by witwolf** [comment]
> > parallel load is not available with SocialBot due to some unknown problems with gazebo load world file
> 
> So are the changes in suite_socialbot.py still necessary if parallel loading is not available? What will be needed to fix the gazebo world file loading issue?

changes in suite_socialbot.py is bug fix for lock handle  ,  i'am trying to solve  gazebo world loading issue (#83) in SocialRobot

**Pull request merged**